hathor/transaction/base_transaction.py

```diff
@@ -844,7 +845,8 @@ def reset_metadata(self) -> None:
         recalculating all metadata.
         """
         assert self.storage is not None
-        self._metadata = TransactionMetadata(hash=self.hash,
+        score = self.weight if self.is_genesis else 0
+        self._metadata = TransactionMetadata(hash=self.hash, score=score,
```

Comment:
Why do we need this change?

Reply:
Same as here the previous comment: https://github.com/HathorNetwork/hathor-core/pull/392/files#r878208656

Reply:
Sure. Thanks!